src/proto/active_packetizer.rs

```diff
@@ -311,23 +310,23 @@ where
                 if xid == 0 && !*this.first {
                     // response to shutdown -- empty response
                     // XXX: in theory, server should now shut down receive end
-                    trace!(logger, "got response to CloseSession");
+                    trace!("got response to CloseSession");
                     if let Some(e) = err {
                         return Poll::Ready(CloseSessionSnafu { error_code: e }.fail());
                     }
                 } else if xid == -1 {
                     // watch event
                     use super::response::ReadFrom;
                     let e = WatchedEvent::read_from(&mut buf)?;
-                    trace!(logger, "got watcher event {:?}", e);
+                    trace!(event = ?e, "got watcher event");
 
                     let mut remove = false;
                     if let Some(watchers) = this.watchers.get_mut(&e.path) {
                         // custom watchers were set by the user -- notify them
                         let mut i = (watchers.len() - 1) as isize;
-                        trace!(logger,
-                               "found potentially waiting custom watchers";
-                               "n" => watchers.len()
+                        trace!(
+                            watchers = watchers.len(),
```

Comment:
suggestion: Rename the field/attribute to make it more clear, this is the _number_ of watchers, not the watchers themself:

```suggestion
                            watcher_count = watchers.len(),
```